### PR TITLE
feat: Add support for COPY from https 

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will make the following SQL work:

Normal URL:

```sql
COPY INTO ontime200 
    FROM 'https://repo.databend.rs/dataset/stateful/ontime_2006_100000.csv' 
    FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)
```

URL with set pattern:

```sql
COPY INTO ontime200 
    FROM 'https://repo.databend.rs/dataset/stateful/ontime_2006_{100000,100001}.csv' 
    FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)
```

URL with range pattern:

```sql
COPY INTO ontime200 
    FROM 'https://repo.databend.rs/dataset/stateful/ontime_2006_[100000-110000].csv' 
    FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)
```

(Powered by [globiter](https://github.com/Xuanwo/globiter))

I added a small test for https. In the future, we will add larger test suites.

Will help to address https://github.com/datafuselabs/databend/pull/6599

### CODE REVIEW

**Comment by Xuanwo** [comment]
> Is it possible to handle such URLs, like `https://www.kaggle.com/datasets/jtentor/blogfeedback-data-set/download?datasetVersionNumber=1`? (In this example it will end up downloading a zip file containing 61 csv files.)

This URL will redirect to

```shell
> GET /datasets/jtentor/blogfeedback-data-set/download?datasetVersionNumber=1 HTTP/2
> Host: www.kaggle.com
> user-agent: curl/7.84.0
> accept: */*
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
< HTTP/2 302
< content-length: 0
< date: Tue, 19 Jul 2022 16:08:41 GMT
< location: /account/login?titleType=dataset-downloads&showDatasetDownloadSkip=False&messageId=datasetsWelcome&returnUrl=%2Fdatasets%2Fjtentor%2Fblogfeedback-data-set%2Fversions%2F1%3Fresource%3Ddownload
< set-cookie: ka_sessionid=ac389d132e1916212e9575fddee8100a; max-age=2626560; path=/
```

Maybe we can support this by allowing users to pass the `cookies` and `authorization` header in `connection` as a setting.

**Comment by Xuanwo** [comment]
> (In this example it will end up downloading a zip file containing 61 csv files.)

Zip support still needs to wait for https://github.com/datafuselabs/opendal/issues/312

**Comment by BohuTANG** [comment]
Have read this PR, found that we need to config a http storage type, why we need this config? Can we copy from https scheme to the others storage type(like s3 or azure blob)? I think this is useful for users to load their data which is on https web, they shouldn't need a https scheme storage type. 🤔

**Comment by Xuanwo** [comment]
> Have read this PR, found that we need to config a http storage type, why we need this config?

Not very useful, this API is served for read only.

> Can we copy from https scheme to the others storage type(like s3 or azure blob)?

Yes.

> I think this is useful for users to load their data which is on https web, they shouldn't need a https scheme storage type.

It does. Let's remove the HTTP storage backend.

File: tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+. "$CURDIR"/../../../shell_env.sh
+
+echo "drop table if exists ontime200;" | $MYSQL_CLIENT_CONNECT
+
+## Create table
+cat $CURDIR/../ddl/ontime.sql | sed 's/ontime/ontime200/g' | $MYSQL_CLIENT_CONNECT
+
+copy_from_location_cases=(
+  # copy csv
+  "copy into ontime200 from 'https://repo.databend.rs/dataset/stateful/ontime_2006_100000.csv' FILE_FORMAT = (type = 'CSV' field_delimiter = ','  record_delimiter = '\n' skip_header = 1)"
```
[the remark this answers is not in the document]
Comment (BohuTANG) [reply tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh:13]: Else LGTM :+1:

[the remark this answers is not in the document]
Comment (BohuTANG) [reply tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh:13]: Oh, can we remove `StorageHttpConfig`?

[the remark this answers is not in the document]
Comment (Xuanwo) [reply tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh:13]: `StorageHttpConfig` is required as an internal storage parameter used in the stage table.

We will not expose this config to users.

Do this look good to you?

[the remark this answers is not in the document]
Comment (BohuTANG) [reply tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh:13]: That's Ok

**Comment by Xuanwo** [comment]
> Can we add more patterns(set and range) here?

Test added, PTAL @BohuTANG

**Review by BohuTANG — APPROVED** [review]

**Comment by BohuTANG** [comment]
By the way, can we support set/range path pattern for s3-like scheme? PATTERN = 'regex_pattern' is not friendly for the user which document in https://databend.rs/doc/reference/sql/dml/dml-copy-into-table

**Review by PsiACE — APPROVED** [review]
lgtm

**Comment by BohuTANG** [comment]
/runner/_work/databend/databend/tests/suites/1_stateful/00_copy_v2/00_0001_copy_from_http_location.sh
[242](https://github.com/datafuselabs/databend/runs/7434253640?check_suite_focus=true#step:4:249)
Error: Process completed with exit code 1.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**